### PR TITLE
Refactor CTMap details into CTMap implementation

### PR DESCRIPTION
**Pull request by joestringer**
Based on ~~#5480~~.

This PR simplifies the callers of the CTMap listing and garbage collection functions to remove duplicated code paths which reimplement common functionality.

Take note particularly for the patch "ctmap: Simplify garbage collection" which changes the ctmap garbage collection functions to be called with a single filter that contains all IPv4/IPv6 addresses to be saved, rather than individually handling IPv4/IPv6 CT maps.

There's also a couple of misc patches here that just refactor code to live in more appropriate places or de-duplicate implementations of, eg, string-formatting.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5488)

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]
NIce

**Review by vadorovsky — APPROVED** [review]

File: pkg/maps/ctmap/ctmap.go [hunk 1]
```diff
@@ -615,3 +626,22 @@ func LocalMaps(e CtEndpoint, ipv4, ipv6 bool) []*Map {
 func GlobalMaps(ipv4, ipv6 bool) []*Map {
 	return maps(nil, ipv4, ipv6)
 }
+
+// WriteBPFMacros writes the map names for conntrack maps into the specified
+// writer, defining usage of the globale map or local maps depending on whether
```
Comment (ianvernon) [inline pkg/maps/ctmap/ctmap.go]: global

**Review by ianvernon — APPROVED** [review]
Awesome!

**Comment by joestringer** [comment]
Green build. Will apply spelling fix in comment.

**Comment by joestringer** [comment]
Given the three positive reviews and green build, it seems reasonable to me to set this as 'ready-to-merge'.

**Pull request merged**